src/main/java/ecdar/backend/EngineConnection.java

```diff
@@ -7,17 +7,21 @@
 import java.util.concurrent.TimeUnit;
 
 public class EngineConnection {
-    private final Process process;
+    private final Engine engine;
     private final EcdarBackendGrpc.EcdarBackendStub stub;
     private final ManagedChannel channel;
-    private final Engine engine;
+    private final Process process;
     private final int port;
 
-    EngineConnection(Engine engine, Process process, EcdarBackendGrpc.EcdarBackendStub stub, ManagedChannel channel) {
-        this.process = process;
+    EngineConnection(Engine engine, ManagedChannel channel, EcdarBackendGrpc.EcdarBackendStub stub) {
+        this(engine, channel, stub, null);
+    }
```

Comment:
Personally, when I use constructor chaining I have the called constructor before. So maybe this constructor should be moved down.

Reply:
The constructors have been swapped :+1: